api/models/author.py

```diff
@@ -60,15 +60,16 @@ def add_pending(self, friend):
         if not self.pending.filter(id=friend.id).exists():
             self.pending.add(friend)
             self.save()
-
         self.follow(friend)
 
     def add_friend(self, friend):
         """Create a friend from an author model"""
-        friend = self._get_cached_author(friend)
-        if not self.friends.filter(id=friend.id).exists():
-            self.friends.add(friend)
+        friend_cached = self._get_cached_author(friend)
+        if not self.friends.filter(id=friend_cached.id).exists():
+            self.friends.add(friend_cached)
             self.save()
+            friend.add_friend(self)
```

Comment:
I would say I am in agreement that we shouldn't be dealing with adding friends to other objects inside this instance. That being said, add_friend should still be a model call. Whatever view designates that two authors should still become friends should be calling this on each respective object.